release: Update version, create release notes and generate changelog (#273)
elastic/ecctl: docs/release_notes/v1.0.0-beta3.adoc

Elastic cloud control (ecctl) is Elastic’s CLI interface to manage Elasticsearch Service (ESS) and Elastic Cloud Enterprise (ECE). It is now open for a public beta.

Download the release binaries:

[%hardbreaks]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_darwin_amd64.tar.gz[ecctl_1.0.0-beta3_darwin_amd64.tar.gz]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_32-bit.deb[ecctl_1.0.0-beta3_linux_32-bit.deb]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_32-bit.rpm[ecctl_1.0.0-beta3_linux_32-bit.rpm]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_386.tar.g[ecctl_1.0.0-beta3_linux_386.tar.g]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_64-bit.deb[ecctl_1.0.0-beta3_linux_64-bit.deb]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_64-bit.rpm[ecctl_1.0.0-beta3_linux_64-bit.rpm]
https://download.elastic.co/downloads/ecctl/1.0.0-beta3/ecctl_1.0.0-beta3_linux_amd64.tar.gz[ecctl_1.0.0-beta3_linux_amd64.tar.gz]

[float]
[id="{p}-release-notes-v1.0.0-beta3-whats-new"]
==== What's new

> * *The `ecctl init` command supports ESS and ESSP users.* You will see a few new options in the init flow. These include the ability to choose your Elastic Cloud offering, and region if you are an Elasticsearch Service customer.

Reviewer: ```suggestion
* *The `ecctl init` command supports Elasticsearch Service and Elasticsearch Service Private users.* You will see a few new options in the init flow: you can now choose your Elastic Cloud offering and the Elasticsearch Service region, if you have a subscription to the service.
```

(Same comment as higher up, AFAIK ESS and ESSP are still not officially approved abbreviations AFAIK, though I did start asking around if that had changed.)

[float]
[id="{p}-release-notes-v1.0.0-beta3-bug-fixes"]
==== Bug fixes

* *Deployment tracking fixed.* Previously the `--track` flag for any command that targeted the deployments API was broken. We are now using a new deployment tracker which also supports AppSearch.

[float]
[id="{p}-release-notes-v1.0.0-beta3-breaking-changes"]
==== Breaking changes

* *Commands which used the deprecated clusters API have been removed.* Actions performed through these commands are available with their `ecctl deployment` or `ecctl deployment resource` counterparts. The following commands and their subcommands have been removed:
+
--
* `ecctl deployment elasticsearch`
* `ecctl deployment kibana`
* `ecctl deployment apm`
* `ecctl deployment appsearch`
--

> * *Deployment create command behavior change.* This command no longer supports overriding json template fields with flags.

Reviewer 2: ```suggestion
* *Deployment create command behavior change.* This command no longer supports overriding JSON template fields with flags.
```
